services/ux-backend/main.go

```diff
@@ -73,12 +83,68 @@ func main() {
 	if err := storagev1.AddToScheme(scheme); err != nil {
 		klog.Exitf("failed to add storagev1 to scheme. %v", err)
 	}
+	if err := noobaaapis.AddToScheme(scheme); err != nil {
+		klog.Exitf("failed to add noobaaapis to scheme. %v", err)
+	}
+
+	klog.Info("Setting up cached client for the ux backend server")
+	clientConfig, err := ctrlconfig.GetConfig()
+	if err != nil {
+		klog.Exitf("failed to get cached client config: %v", err)
+	}
+
+	cache, err := ctrlcache.New(clientConfig, ctrlcache.Options{
+		Scheme: scheme,
+		DefaultNamespaces: map[string]ctrlcache.Config{
+			namespace:                    {},
+			"openshift-storage-extended": {},
+		},
+	})
+	if err != nil {
+		klog.Exitf("failed to create cache: %v", err)
+	}
+
+	if err := cache.IndexField(ctx, &storagev1.StorageClass{}, "provisioner", func(obj client.Object) []string {
+		sc := obj.(*storagev1.StorageClass)
+		return []string{sc.Provisioner}
+	}); err != nil {
+		klog.Exitf("failed to add provisioner index for StorageClasses: %v", err)
+	}
 
```

Comment:
I agree an index for provisioner isn't mandatory and can simply loop over at call site.

Reply:
ack, I will remove indexing on StorageClass then. Thanks for confirming.

Reply:
no, it is not a prefix, that's my point in https://github.com/red-hat-storage/ocs-operator/pull/3536#discussion_r2477182388 as mentioned in the linked comment, did you ask either RGW or MCG folks to confirm? I believe they are not changing it, however better confirm once.